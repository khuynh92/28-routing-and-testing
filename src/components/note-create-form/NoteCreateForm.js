import React, { Component } from 'react';

export default class NoteCreateForm extends Component {
  render() {
    return (
      <form onSubmit={this.props.handleSubmit}>
        <label>
          <input value={this.props.text} onChange={this.props.handleChange} id='note' type="text"/>
          <input type="submit"/>
        </label>
      </form>
    );
  }
}