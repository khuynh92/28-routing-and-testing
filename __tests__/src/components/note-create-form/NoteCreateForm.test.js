import React from 'react';
import renderer from 'react-test-renderer';

import NoteCreateForm from '../../../../src/components/note-create-form/NoteCreateForm.js';

describe('<Dashboard/> Enzyme Test', () => {
  it('loads components at application start', () => {
    let app = shallow(<NoteCreateForm/>);
    expect(app.find('form').exists()).toBeTruthy();
  });
});

describe('<Dashboard/> Snapshot Test', () => {
  it('renders correctly', () => {
    const component = renderer.create(<NoteCreateForm />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});


