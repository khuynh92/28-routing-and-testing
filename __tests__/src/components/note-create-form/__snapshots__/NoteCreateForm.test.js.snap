// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> Snapshot Test renders correctly 1`] = `
<form>
  <label>
    Title:
    <br />
    <input
      id="title"
      type="text"
    />
  </label>
  <br />
  <label>
    Notes:
    <br />
    <input
      id="content"
      type="text"
    />
  </label>
  <input
    type="submit"
  />
</form>
`;
